packages/camera/camera_windows/windows/camera_plugin.cpp

```diff
@@ -341,6 +369,53 @@ void CameraPlugin::StopVideoRecording(
   }
 }
 
+void CameraPlugin::StartImageStream(
+    int64_t camera_id,
+    std::function<void(std::optional<FlutterError> reply)> result) {
+  // check if request already exists
+  Camera* camera = GetCameraByCameraId(camera_id);
+  if (!camera) {
+    return result(FlutterError("camera_error", "Camera not created"));
+  }
+  if (camera->HasPendingResultByType(PendingResultType::kStartStream)) {
+    return result(
+        FlutterError("camera_error", "Pending start stream request exists"));
+  }
+
+  if (!event_sink) {
+    return result(FlutterError("camera_error",
+                               "Unable to make event channel from windows"));
+  }
+
+  if (camera->AddPendingVoidResult(PendingResultType::kStartStream,
```

Comment:
Where are the pending results added in this PR resolved? I don't see anything in the diff that retrieves them.

Reply:
good question, when i wrote it figured id deal with that later and forgot. What exactly are these checks used for and are they necessary if the function is synchronous?

Reply:
There will be no result object if the API is made synchronous.